pkg/operator/k8sutil/cmdreporter/cmdreporter.go

```diff
@@ -92,22 +92,10 @@ func New(
 	cmd, args []string,
 	rookImage, runImage string,
 ) (*CmdReporter, error) {
-	if clientset == nil || ownerRef == nil {
-		return nil, fmt.Errorf("clientset [%+v] and owner reference [%+v] must be specified", clientset, ownerRef)
-	}
-	if appName == "" || jobName == "" || jobNamespace == "" {
-		return nil, fmt.Errorf("app name [%s], job name [%s], and job namespace [%s] must be specified", appName, jobName, jobNamespace)
-	}
-	// at least one command must be set, and it cannot be an empty string
-	if len(cmd) == 0 || cmd[0] == "" {
-		return nil, fmt.Errorf("command [%+v] must be specified", cmd)
-	}
-	if rookImage == "" || runImage == "" {
-		return nil, fmt.Errorf("Rook image [%s] and run image [%s] must be specified", rookImage, runImage)
-	}
 	cfg := &cmdReporterCfg{
 		clientset:    clientset,
 		ownerRef:     ownerRef,
+		appName:      appName,
```

Comment:
What's the relationship between the `cfg` properties and the parameters? Seems like we should use the same ones that we're validating in the error checking.

Reply:
I think this is answered by Elise's comment above?

Reply:
Ok, I'm understanding the validation changes, now one more question... The original bug was that an empty `appName` was allowed, right? Why didn't the validation on line 98 catch this? `if appName == ""`. The error checking looks effectively the same to me to check the parameters as before, or to validate the struct after it's created.

Reply:
Oh, I see now. `appName` just wasn't be passed to the cfg struct previously.